This tutorial will continue using the `cathedral` and `GSS` datasets. Let's load the datasets first.

---



In [1]:
install.packages("plyr")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
library(tidyverse)
library(ggplot2)
library(plyr)

cathedral <- readRDS(url('https://github.com/omerfyalcin/colab-data/blob/main/cathedral.rds?raw=true'))
GSS <- read_csv(url('https://raw.githubusercontent.com/RosemaryPang/Data-for-teaching/main/gss2021.csv'))


── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
------------------------------------------------------------------------------

You have loaded plyr after dplyr - this is likely to cause problems.
If you need functions from both plyr and dplyr, please load plyr first, then dplyr:
library(plyr); library(dplyr)

------------------------------------------------------------------------------


Attaching package: ‘plyr’


The following objects are masked from ‘package:dplyr’:

    arrange, count, desc, failwith, id, mutate,

## **Hypothesis Testing**  
  
Hypothesis testing is strongly tied to the concept of confidence intervals.
For example, if a hypothesized population mean $\mu$ is not within the 95% confidence interval, then we can reject the null hypothesis that the population mean equals $\mu$ at the 5% significance level.

Elements of a Statistical Test:  
1. Null Hypothesis, $H_{0}$  
2. Alternative Hypothesis, $H_{\alpha}$  
3. Test Statistics  
4. Rejection Region
5. Conclusion  


### **`Null Hypothesis`**
  
The null hypothesis, denoted $H_{0}$, is a statement to be tested. The null hypothesis is usually a statement of no change, no effect or no difference and is assumed true until evidence indicates otherwise. The parameter value of $H_{0}$ is a single point (because of the equality sign).
  
### **`Alternative Hypothesis`**
  
The alternative hypothesis, denoted by $H_{\alpha}$ or $H_{1}$, is a statement that we are trying to find evidence (from the sample data) to support. The parameter values of $H_{\alpha}$ form a region, i.e. there are multiple values (because of inequality signs). Alternative Hypothesis can either be one-sided {i.e. either left-tailed or right tailed} or two-sided.
  
### **`Test Statistics`**
  
The Test statistic is a function of the sample measurements on which the statistical decision will be based. It usually contains data such as significance level, mean, median, standard deviation, standard error, type of distribution used to make calculations.
  
### **`Rejection Region`**
  
The Rejection Region `(RR)`, specifies the values of the test statistic for which the null hypothesis will be rejected in favor of the alternate hypothesis. In other words, if the test statistic falls in the `RR`, then we reject the null hypothesis. We also determine the $p-value$ in Rejection Region. `p-value` is the probability that test statistic equals the observed data or data even more extreme as predicted by $H_{\alpha}$. It is calculated by assuming $H_{0}$ to be true.  
  
### **`Conclusion`**  
  
After conducting a series of tests and calculating values from distributions, p-value, and test statistics, we should be able to reject or fail to reject the null hypothesis based on feedback and insights from your sample data.  
  
If our `p-value` is less than the pre-determined significance ($\alpha$) level, typically **5%**, we reject null hypothesis and conclude that we do have statistical evidence for the alternative hypothesis is true. If our `p-value` is greater than the significance level, we fail to reject null hypothesis. Hence, we do not have enough statistical evidence to draw conclusions.

### **`One Sample t-test`**  
  
The `t.test()` function has another additional argument: `mu`. `mu` determines a number indicating the true value of the mean (or a difference in means if it is a two sampled test). The default value is 0 but you can specify whatever you want based on the needs of your application.    
  
Let's test whether the population mean Height of Cathedral is 85 ($H_0: \mu = 85$) against the alternative hypothesis that it is not ($H_a: \mu \neq 85$).  

In [3]:
t.test(cathedral$Height, mu = 85)


	One Sample t-test

data:  cathedral$Height
t = -3.2778, df = 24, p-value = 0.00318
alternative hypothesis: true mean is not equal to 85
95 percent confidence interval:
 68.96409 81.35591
sample estimates:
mean of x 
    75.16 


After executing this code, we can see that the `t.test()` function displays information about the alternative hypothesis. For the above code, we can see that we have a very small p-value, which indicates that we can reject the null hypothesis at most conventional levels, such as 5% or 1%, in favor of the alternative that the true mean is not 85. The 95% confidence interval is automatically calculated. The fact that it does not contain 85 shows the same thing another way. In the `t.test()` function, we can change the alternative hypothesis to test on the specified data by adding another argument: `alternative`. `alternative` by default is `two.sided` but it's values can either be `greater`, `less`, or `two.sided`.  
  
Let's test whether the true mean of the Height population of Cathedral is less than 85.

In [4]:
t.test(cathedral$Height, mu = 85, alternative = 'less')


	One Sample t-test

data:  cathedral$Height
t = -3.2778, df = 24, p-value = 0.00159
alternative hypothesis: true mean is less than 85
95 percent confidence interval:
     -Inf 80.29614
sample estimates:
mean of x 
    75.16 


Note that, when we use 'less', the p-value is half of what it was when we used 'two.sided.' This is because, in a one sided test, only extreme values only on one side of the null distribution are considered evidence against the null. Intuitively, we have a sample with a mean of 75.16 and we are considering whether the population mean is 85. We are more confident when concluding that the population must be lower than 85 (a smaller p-value) than when concluding that the population mean is not 85 (a still small, but somewhat larger p-value than the former). For the latter, deviations in either direction could be evidence.
  
Your turn: Now  test whether the alternative that true mean of Height of Cathedral is greater than 85.

In [5]:
# Write your code here
t.test(cathedral$Height, mu = 85, alternative = 'greater')


	One Sample t-test

data:  cathedral$Height
t = -3.2778, df = 24, p-value = 0.9984
alternative hypothesis: true mean is greater than 85
95 percent confidence interval:
 70.02386      Inf
sample estimates:
mean of x 
    75.16 



Since the p-value (0.9984) is much greater than the typical significance level of 0.05, we fail to reject the null hypothesis. There is not enough statistical evidence to support the claim that the true mean height of the cathedral is greater than 85.

Correct answer: Fail to reject the null hypothesis

What conclusions would you draw about the hypothesis after examining the solutions? What do you make of the p-value? Are you able to reject the null in favor of the alternative hypothesis?

### `prop.test()`  
  
`prop.test()` is the **Z-Test of Proportions with Continuity Correction**. It is similar to the t-test, but is suited for tests about proportions. `prop.test()` has 3 main arguments:`x`, `n`, and `p`.
There can be additional requirements such as:`alternative`, `conf.level`, and `correct`.  
  
Here,  
1. `x`: vector of counts of successes
2. `n`: vector of counts of trials  
3. `p`: vector of probabilities of success  
4. `alternative`: either `two.sided`, `greater`, or `less`
5. `conf.level`: denotes the confidence level chosen  
6. `correct`: a logical value denoting where to apply continuity correction

Let's do a simulated example:

In [6]:
set.seed(0)
heads <- rbinom(1, size = 100, p = 0.3)
prop.test(heads, 100)


	1-sample proportions test with continuity correction

data:  heads out of 100, null probability 0.5
X-squared = 2.89, df = 1, p-value = 0.08913
alternative hypothesis: true p is not equal to 0.5
95 percent confidence interval:
 0.3140323 0.5129652
sample estimates:
   p 
0.41 


Even though we simulated with a probability of heads equal to 0.3 (30%), the 95% percent confidence interval for the prop.test() does include 0.5. We are unable to reject the null hypothesis that the true probability is 0.5 (which is the default null hypothesis in a one sample test in `prop.test()`) at the 5% significance level.

Your turn: Even though the underlying probability was 0.3, the prop.test() could not rule out that it's actually 0.5. Maybe it's because of the sample size. Increase the sample size to 200 to see if you'll get a different result.

In [7]:
# Write your code here
set.seed(0)
heads_200 <- rbinom(1, size = 200, p = 0.3)
prop.test(heads_200, 200)


	1-sample proportions test with continuity correction

data:  heads_200 out of 200, null probability 0.5
X-squared = 61.605, df = 1, p-value = 4.198e-15
alternative hypothesis: true p is not equal to 0.5
95 percent confidence interval:
 0.1659406 0.2850661
sample estimates:
   p 
0.22 
